### DISCUSSION TITLE
Is there a setting that allows messages not to be removed from the stream in the case of term()?

### DISCUSSION THREAD

**Discussion by AlexandrTQ** [nats-io/nats-server#5110] [Q&A]
When I read an event from subject $JS.EVENT.ADVISORY.CONSUMER.MAX_DELIVERIES.<STREAM>.<CONSUMER>, I can get the original message. 
When I try to do the same for the $JS.EVENT.ADVISORY.CONSUMER.MSG_TERMINATED.<STREAM>.<CONSUMER> event, i get 404 when trying to get the original message.
According to the documentation: https://docs.nats.io/using-nats/developer/develop_jetstream/consumers.
_"If you also need to have access to the message in question then you can use the message's sequence number included in the advisory to retrieve that specific message by sequence number from the stream. If a message reaches its maximum level of delivery attempts, it will still stay in the stream until it is manually deleted or manually acknowledged."_

As I understand it, by default, only messages that have reached max_delivered remain in the stream. Is it possible to prevent term() messages from being deleted in the same way?

**Comment by ripienaar** [comment]
It depends on your stream retention policy, what did you set it to? Show `nats stream info`

**Reply by AlexandrTQ** [reply]
{
    "config": {
        "name": "test_stream",
        "subjects": [
            "subject_1",
            "subject_2"
        ],
        "retention": "workqueue",
        "max_consumers": -1,
        "max_msgs": -1,
        "max_bytes": -1,
        "discard": "old",
        "max_age": 0,
        "max_msgs_per_subject": -1,
        "max_msg_size": -1,
        "storage": "file",
        "num_replicas": 1,
        "duplicate_window": 60000000000,
        "compression": "none",
        "allow_direct": false,
        "mirror_direct": false,
        "consumer_limits": {}
    },
}

I tried switching retention to LimitsPolicy and yes, term() messages stopped being deleted, but I also don't want Ask() messages to be retained.

**Reply by ripienaar** [reply]
That's how it works atm, terminate is considered acked so it removes it
